### PR TITLE
Post physics transforms

### LINKED ISSUE #614: Update entity position after physics is done
When physics is complete, set world position and rotation of owning entity. This should make meshes render in the correct place according to physics feedback.

### PR DESCRIPTION
**Pull request by andens**
Resolves #614

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![INFO][INFO] [EntityEditor.cpp#L444](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2cf3b5180e7fd46440bff7d160b0d7f6846b03f8/src/Editor/GUI/Editors/EntityEditor.cpp#L444): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Component/RigidBody.cpp [hunk 1]
```diff
@@ -41,23 +40,37 @@ namespace Component {
...
 
     void RigidBody::Position(const glm::vec3& pos) {
-        btTransform trans;
-        rigidBody->getMotionState()->getWorldTransform(trans);
+        btTransform trans = rigidBody->getWorldTransform();
         trans.setOrigin(Physics::glmToBt(pos));
         rigidBody->setWorldTransform(trans);
     }
 
+    glm::quat RigidBody::Orientation() const {
```
Comment (Chainsawkitten) [inline src/Engine/Component/RigidBody.cpp]: I think GetOrientation and SetOrientation would be clearer than two methods with that same name that do different things.

File: src/Engine/Component/RigidBody.cpp [hunk 2]
```diff
@@ -41,23 +40,37 @@ namespace Component {
...
+        return Physics::btToGlm(trans.getRotation());
+    }
+
+    void RigidBody::Orientation(const glm::quat& rotation) {
+        btTransform trans = rigidBody->getWorldTransform();
+        trans.setRotation(Physics::glmToBt(rotation));
+        rigidBody->setWorldTransform(trans);
+    }
+
+    float RigidBody::Mass() {
```
Comment (Chainsawkitten) [inline src/Engine/Component/RigidBody.cpp]: Same as with Orientation.

File: src/Engine/Component/RigidBody.cpp [hunk 3]
```diff
@@ -40,24 +39,38 @@ namespace Component {
...
-        btTransform trans;
-        rigidBody->getMotionState()->getWorldTransform(trans);
+    void RigidBody::SetPosition(const glm::vec3& pos) {
+        btTransform trans = rigidBody->getWorldTransform();
         trans.setOrigin(Physics::glmToBt(pos));
         rigidBody->setWorldTransform(trans);
     }
 
-    void RigidBody::Mass(float mass) {
+    glm::quat RigidBody::GetOrientation() const {
```
Comment (NinjaDanz3r) [inline src/Engine/Component/RigidBody.cpp]: MINOR The function 'GetOrientation' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

File: src/Engine/Component/RigidBody.cpp [hunk 4]
```diff
@@ -40,24 +39,38 @@ namespace Component {
         }
     }
 
-    glm::vec3 RigidBody::Position() const {
-        btTransform trans;
-        rigidBody->getMotionState()->getWorldTransform(trans);
+    glm::vec3 RigidBody::GetPosition() const {
```
Comment (NinjaDanz3r) [inline src/Engine/Component/RigidBody.cpp]: MINOR The function 'GetPosition' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![INFO][INFO] [EntityEditor.cpp#L444](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/23254cdc147b1b5f4fe9b0e75cc04d7d27984601/src/Editor/GUI/Editors/EntityEditor.cpp#L444): keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**